app/src/main/java/co/kr/bemyplan/ui/purchase/after/adapter/DailyContentsAdapter.kt

```diff
@@ -1,38 +1,97 @@
 
-    private val dailyContentsList = mutableListOf<DailyContents>()
-    private val spotList = mutableListOf<Spot>()
+    private var spotList = listOf<Spot>()
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DailyContentsViewHolder {
-        val binding = DailyContentsListBinding.inflate(
-            LayoutInflater.from(parent.context),
-            parent, false
-        )
+    override fun getItemViewType(position: Int) = viewType
 
-        return DailyContentsViewHolder(binding)
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = when(viewType) {
+        TYPE_CONTENTS -> {
+            val binding = ItemDailyContentsBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent, false
+            )
+            ContentsViewHolder(binding)
+        }
+        TYPE_ROUTE -> {
+            val binding = ItemDailyRouteBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent, false
+            )
+            RouteViewHolder(binding)
+        }
+        else -> {
+            throw IllegalStateException("Not Found ViewHolder Type $viewType")
+        }
     }
 
     override fun onBindViewHolder(holder: DailyContentsViewHolder, position: Int) {
-        holder.onBind(dailyContentsList[position])
+        when(holder) {
+            is ContentsViewHolder -> {
+                holder.onBind(spotList[position])
+            }
+            is RouteViewHolder -> {
+                holder.onBind(spotList[position])
+            }
+        }
     }
 
-    override fun getItemCount() = dailyContentsList.size
+    override fun getItemCount() = spotList.size
+
+    @SuppressLint("NotifyDataSetChanged")
+    fun setItems(items: List<Spot>) {
+        spotList = items
+        notifyDataSetChanged()
+    }
+
+    // MVVM을 위한 코드
+   @BindingAdapter("bind:spotList")
+    fun bindSpotList(recyclerView: RecyclerView, items: ObservableArrayList<Spot>) {
+        val adapter = recyclerView.adapter as? DailyContentsAdapter
+        adapter?.setItems(items)
+    }
 
-    class DailyContentsViewHolder(private val binding: DailyContentsListBinding)
+    abstract class DailyContentsViewHolder(private val binding: ViewDataBinding)
         : RecyclerView.ViewHolder(binding.root) {
-            fun onBind(data: DailyContents) {
-                binding.tvPlace.text = data.placeName[0]
-                binding.tvContext.text = data.context[0]
-                binding.tvAddress.text = data.address[0]
-            }
+        abstract fun onBind(data: Spot)
+    }
+
+    class ContentsViewHolder(private val binding: ItemDailyContentsBinding): DailyContentsViewHolder(binding) {
+        override fun onBind(data: Spot) {
+            binding.spot = data
+            Glide.with(binding.root)
+                .load(data.photo)
+                .apply(RequestOptions.bitmapTransform(RoundedCorners(5)))
+                .into(binding.ivPhoto)
```

Comment:
```suggestion
        with(binding) { 
        spot = data
             Glide.with(root)
                 .load(data.photo)
                  .apply(RequestOptions.bitmapTransform(RoundedCorners(5)))
                  .into(ivPhoto)
                  }
     }
```
스코프 함수 쓰면 좀 더 간결하게 표현이 가능할듯 합니당~!!

Reply:
감사합니다. 스코프함수 메모...